Fix type assert from int to int64.

vivekmenezes opened the pull request.

vivekmenezes committed:
Fix type assertion from int to int64. Added unittests.

tschottdorf commented:
LGTM after squashing the first commit into the second.  The commit message could mention your test update as well.

vivekmenezes commented:
Changed this a bit to be safer, and addressed your comment.

On Fri, Aug 7, 2015 at 8:07 AM Tobias Schottdorf <notifications@github.com>
wrote:

> LGTM after squashing the first commit into the second. The commit message
> could mention your test update as well.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/2017#issuecomment-128684090>
> .
>

petermattis commented on sql/server.go:
```diff
@@ -139,8 +144,9 @@ func (p parameters) Arg(i int) (parser.Datum, bool) {
 	case *string:
 		return parser.DString(*t), true
+	default:
+		panic(fmt.Sprintf("Incorrect type %T", t))
```
`s/Incorrect/unexpected/g`?

vivekmenezes replied:
I'll incorporate all the feedback and fix the commented out test tonight.

On Fri, Aug 7, 2015, 8:41 AM Peter Mattis <notifications@github.com> wrote:

> In sql/server.go
> <https://github.com/cockroachdb/cockroach/pull/2017#discussion_r36514401>:
>
> > @@ -139,8 +144,9 @@ func (p parameters) Arg(i int) (parser.Datum, bool) {
> >  		return parser.DString(t), true
> >  	case *string:
> >  		return parser.DString(*t), true
> > +	default:
> > +		panic(fmt.Sprintf("Incorrect type %T", t))
>
> s/Incorrect/unexpected/g?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/2017/files#r36514401>.
>
